src/Compilers/CSharp/Test/Syntax/Parsing/NullableParsingTests.cs

```diff
@@ -717,15 +717,15 @@ public void DeclarationPattern_NullableType()
         public void DeclarationPattern_NullableArray()
         {
             UsingStatement("switch (e) { case T[]? t: break; }",
-                // (1,21): error CS0443: Syntax error; value expected
+                // (1,19): error CS8652: The feature 'type pattern' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
```

Comment:
This seems worrisome. I thought we disambiguated `?` by saying that it wasn't part of a type pattern, and thus it would parse to `?:` in legacy cases. Should this still be a `?:`?

Reply:
The change here is caused by T[]? previously being treated as the start of an expression starting with an index operator.  Now we disambiguate it as a type pattern.  Either way, the `?` isn't part of the pattern, but is an unexpected token that results in a syntax error.  Both interpretations result in a syntax error, so we prefer the type interpretation (which is the change)

---
In reply to: [354611441](https://github.com/dotnet/roslyn/pull/39913#discussion_r354611441) [](ancestors = 354611441)